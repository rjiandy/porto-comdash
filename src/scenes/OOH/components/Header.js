// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';

import {View, Text} from '../../../general/components/coreUIComponents';
import {PlaceholderView} from '../../../general/components/UIComponents';
import commaSeparator from '../../../general/helpers/commaSeparator';
import {THEME_COLOR, GREY} from '../../../general/constants/colors';
import {HEADER_FONT_SIZE} from '../../../general/constants/text';

import type {OOH} from '../type/OOH-type';

type Props = {
  data: ?OOH;
  selectedTerritory: string;
};

export default function Header(props: Props) {
  let {data, selectedTerritory} = props;
  if (!selectedTerritory) {
    return (
      <View style={{height: 80}}>
        <PlaceholderView text="Please select territory first" />
      </View>
    );
  } else if (!data) {
    return (
      <View style={{height: 80}}>
        <PlaceholderView
          text={`No data available for territory ${selectedTerritory}`}
        />
      </View>
    );
  }

  return (
    <View style={styles.header}>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.billboard || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Billboard
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.baliho || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Baliho
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.minibillboard || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Mini Billboard
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.streetsignage || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Street Sign
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.others || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Other OOH
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  headerItem: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    flexDirection: 'row',
    borderWidth: 5,
    borderColor: THEME_COLOR,
    borderStyle: 'solid',
    paddingVertical: 10,
    paddingHorizontal: 30,
  },
  headerData: {
    color: GREY,
    fontWeight: 'bold',
  },
  headerDataNumber: {
    fontSize: HEADER_FONT_SIZE + 10,
  },
});
