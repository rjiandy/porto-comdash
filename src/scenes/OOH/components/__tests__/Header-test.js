// @flow

import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from '../Header';

it('should render Header component correctly', () => {
  let data = {
    territory: 'Ambon',
    billboard: 31,
    baliho: 20,
    minibillboard: 30,
    streetsignage: 15,
    others: 11,
  };
  let rendered = shallow(<Header data={data} selectedTerritory="Bali" />);

  expect(toJSON(rendered)).toMatchSnapshot();
});
