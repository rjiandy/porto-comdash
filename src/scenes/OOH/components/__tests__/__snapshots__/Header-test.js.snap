// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<View
  style={126}
>
  <View
    style={125}
  >
    <TextComponent
      customStyle="header"
      style={
        Array [
          127,
          128,
        ]
      }
    >
      31
    </TextComponent>
    <TextComponent
      customStyle="title"
      style={127}
    >
      Billboard
    </TextComponent>
  </View>
  <View
    style={125}
  >
    <TextComponent
      customStyle="header"
      style={
        Array [
          127,
          128,
        ]
      }
    >
      20
    </TextComponent>
    <TextComponent
      customStyle="title"
      style={127}
    >
      Baliho
    </TextComponent>
  </View>
  <View
    style={125}
  >
    <TextComponent
      customStyle="header"
      style={
        Array [
          127,
          128,
        ]
      }
    >
      30
    </TextComponent>
    <TextComponent
      customStyle="title"
      style={127}
    >
      Mini Billboard
    </TextComponent>
  </View>
  <View
    style={125}
  >
    <TextComponent
      customStyle="header"
      style={
        Array [
          127,
          128,
        ]
      }
    >
      15
    </TextComponent>
    <TextComponent
      customStyle="title"
      style={127}
    >
      Street Sign
    </TextComponent>
  </View>
  <View
    style={125}
  >
    <TextComponent
      customStyle="header"
      style={
        Array [
          127,
          128,
        ]
      }
    >
      11
    </TextComponent>
    <TextComponent
      customStyle="title"
      style={127}
    >
      Other OOH
    </TextComponent>
  </View>
</View>
`;
